### DISCUSSION TITLE
How do I get textual app restarted after running a subprocess

### DISCUSSION THREAD

**Discussion by WaylonWalker** [Textualize/textual#1654] [Q&A]
I would like to build a textual app to explore objects, I want the ability to edit them with the users text editor of choice.  I put a POC into the tutorial, I am able to open the editor in a subprocess, and refresh textual, but I am unable to get textual to respond after the editor is closed.

Here is my POC, at the action_edit

https://github.com/WaylonWalker/textual-tutorial/blob/8b7e27d5c68cc69fcea3a8a2bccde215e6917589/textual_tutorial/tui.py#L118

I recorded a short clip of it running.  I hit the edit key around 0:27 and after the editor is closed, textual is unresponsive.

[textual-locked.webm](https://user-images.githubusercontent.com/22648375/214323847-c38c5903-ff6f-454e-a41e-2b8ab7e41ee3.webm)

**Comment by WaylonWalker** [comment]
nevermind, found the issue

https://github.com/Textualize/textual/discussions/165

**Comment by davep** [comment]
If I'm understanding your question correctly, I think [this PR that is in the works right now](https://github.com/Textualize/textual/pull/1655) will help here (once live in Textual).

**Reply by WaylonWalker** [reply]
Awesome, Thanks Dave!  Textual is moving quick!

**Reply by davep** [reply]
Most welcome. It needs a bit more testing and of course an internal review but I'm hopeful it'll be part of the next release. We can mainly thank @JoshKarpel for that PR.

**Comment by WaylonWalker** [comment]
Not sure if this is correct, but I was able to get control back with stop_application_mode and start_application_mode

```python
    def action_edit(self):
        prompt = {"prompt": "this is the prompt"}
        editor = os.environ.get("EDITOR", "vim")
        file = tempfile.NamedTemporaryFile(prefix="lockhart", suffix=".toml")
        file.write(tomlkit.dumps(prompt).encode())
        file.seek(0)
        self._driver.stop_application_mode()
        with subprocess.Popen([editor, file.name]) as proc:
            proc.wait()
        self.refresh()
        self._driver.start_application_mode()
```

**Reply by davep** [reply]
That's roughly correct, as far as I'm aware, although the changes that are  hopefully coming will make this a lot more tidy and will handle a couple of other issues.
